From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Mon, 22 May 2017 21:08:08 +0200
Subject: net/core: remove explicit do_softirq() from busy_poll_stop()
Origin: https://www.kernel.org/pub/linux/kernel/projects/rt/4.11/older/patches-4.11.9-rt7.tar.xz

Since commit 217f69743681 ("net: busy-poll: allow preemption in
sk_busy_loop()") there is an explicit do_softirq() invocation after
local_bh_enable() has been invoked.
I don't understand why we need this because local_bh_enable() will
invoke do_softirq() once the softirq counter reached zero and we have
softirq-related work pending.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 net/core/dev.c |    2 --
 1 file changed, 2 deletions(-)

--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -5060,8 +5060,6 @@ static void busy_poll_stop(struct napi_s
 	if (rc == BUSY_POLL_BUDGET)
 		__napi_schedule(napi);
 	local_bh_enable();
-	if (local_softirq_pending())
-		do_softirq();
 }
 
 bool sk_busy_loop(struct sock *sk, int nonblock)
